src/components/Home/EventCard/index.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-underscore-dangle */
+import React, { Component } from 'react';
+import {
+  Text, View, TouchableOpacity, ImageBackground,
+} from 'react-native';
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+import NavigationWithoutProps from 'utils/navigationWithoutProps';
+import CustomFormat from 'utils/customFormat';
+import styles from './index.styles';
+
+class EventCard extends Component {
+  goToEventDetail = () => {
+
+  }
+
+  goToQR = () => {
+    NavigationWithoutProps.navigate('QR');
+  }
+
+  render() {
+    const { item } = this.props;
+    const time = new Date(item.event.start_time);
+    const calendarText = CustomFormat.formatFullDate(time);
+    let ribbonColor;
+    let ribbonText;
+    switch (item.name) {
+      case 'admin':
+        ribbonColor = 'green';
+        ribbonText = 'white';
+        break;
+      case 'operator':
+        ribbonColor = 'yellow';
+        ribbonText = 'black';
+        break;
+      case 'staff':
+        ribbonColor = 'orange';
+        ribbonText = 'white';
+        break;
+      case 'custom admin':
+        ribbonColor = 'white';
+        ribbonText = 'black';
+        break;
+      default:
+        break;
+    }
```

Comment:
write a method to map name <=> color and text

Reply:
Okay, I have done it